airflow-core/src/airflow/ui/src/utils/datetimeUtils.ts

```diff
@@ -17,9 +17,36 @@
  * under the License.
  */
 import dayjs from "dayjs";
+import dayjsDuration from "dayjs/plugin/duration";
 
-export const getDuration = (startDate: string | null, endDate: string | null) =>
-  dayjs
-    .duration(dayjs(endDate ?? undefined).diff(startDate ?? undefined))
-    .asSeconds()
-    .toFixed(2);
+dayjs.extend(dayjsDuration);
+
+export const getDuration = (startDate?: string | null, endDate?: string | null) => {
+  const segments = [];
+
+  const seconds = dayjs.duration(dayjs(endDate ?? undefined).diff(startDate ?? undefined)).asSeconds();
+
+  if (seconds < 10) {
+    return `${seconds.toFixed(2)}s`;
+  }
+
+  const day = Math.floor(seconds / 60 / 60 / 24);
```

Comment:
dayjs offers formatting seconds to required format. Perhaps this logic could be simplified as below to utilize this.

Ref : https://stackoverflow.com/a/71004873/2610955

```js
if (seconds < 86400) {
    return dayjs.duration(seconds, "seconds").format('HH:mm:ss');
} else {
    return dayjs.duration(seconds, "seconds").format('D[d]HH:mm:ss');
}
```

Reply:
I like that!